### PR TITLE
DEV: Fix Flake8 Errors And Typos In `unet.py`

### PR DESCRIPTION
**Pull request by shaneahmed**
- Fix flake8 errors and typos in `unet.py`

### CODE REVIEW

File: tiatoolbox/models/architecture/unet.py [hunk 1]
```diff
@@ -217,16 +215,18 @@ def __init__(
...
+        if encoder not in ["resnet50", "unet"]:
+            raise ValueError(f"Unknown encoder `{encoder}`")
+
         if encoder == "resnet50":
             padding = 1
             preact = True
             self.backbone = ResNetEncoder.resnet50(num_input_channels)
-        elif encoder == "unet":
+
+        if encoder == "unet":
```
Comment (mostafajahanifar) [inline tiatoolbox/models/architecture/unet.py:226]: This part has already been done in #273 as well as some other bits

**Review by mostafajahanifar — CHANGES_REQUESTED** [review]
Some of these fixes have been already done in #273 and that PR is long time waiting to be merged. I would suggest first merging that PR and then come back to this one.

**Comment by shaneahmed** [comment]
@mostafajahanifar and @John-P  this is ready for review.

File: tiatoolbox/models/architecture/unet.py [hunk 2]
```diff
@@ -168,11 +167,62 @@ def forward(self, x: torch.Tensor):
...
 
 
+def create_block(preact, kernels, input_ch, output_ch):
+    """Helper to create a block of Vanilla Convolution.
+
+    This is in pre-activation style.
+
+    Args:
+        preact (bool):
+            Requires preactivation layers.
```
Comment (mostafajahanifar) [inline tiatoolbox/models/architecture/unet.py:177]: Whether to apply activation layer before the convolution layer. Should be True for ResNet blocks.

**Review by mostafajahanifar — APPROVED** [review]
All looks fine Shan, Thanks.

Just added a small comment if you like you can ignore.

File: tiatoolbox/models/architecture/unet.py (hunk 2 above)
Comment (shaneahmed) [inline tiatoolbox/models/architecture/unet.py:177]:
```suggestion
            Whether to apply activation layer before the convolution layer. Should be True for ResNet blocks.
```

Comment (shaneahmed) [reply tiatoolbox/models/architecture/unet.py]: Addressed in this commit https://github.com/TissueImageAnalytics/tiatoolbox/pull/345/commits/174ea015e1861c4c40877a3d25f70d30f3bed479

File: tiatoolbox/models/architecture/unet.py [hunk 3]
```diff
@@ -256,62 +307,14 @@ def __init__(
...
-
         self.uplist = nn.ModuleList()
         for ch_idx, ch in enumerate(down_ch_list[1:]):
             next_up_ch = ch
             if ch_idx + 2 < len(down_ch_list):
                 next_up_ch = down_ch_list[ch_idx + 2]
             if self.skip_type == "concat":
                 ch *= 2
-            layers = create_block(decoder_block, ch, next_up_ch)
+            layers = create_block(preact, decoder_block, ch, next_up_ch)
```
Comment (John-P) [inline tiatoolbox/models/architecture/unet.py:317]: What is 'preact'?

Comment (John-P) [reply tiatoolbox/models/architecture/unet.py:317]: If it is pre activation, could we just call it that, i.e. `pre_activation`? To make it more readable.

**Review by John-P — APPROVED** [review]

**Comment by John-P** [comment]
Just one minor comment on clarifying an unclear variable name

File: tiatoolbox/models/architecture/unet.py (hunk 3 above)
Comment (shaneahmed) [reply tiatoolbox/models/architecture/unet.py]: Resolved in this commit https://github.com/TissueImageAnalytics/tiatoolbox/pull/345/commits/393fe26d121ef21f28b56e0894bf14e69e98b0ec

**Pull request merged**